debRelease.rb

```diff
@@ -93,30 +94,34 @@ def get_package(component, architecture)
     paths = paths_exist + (paths - paths_exist)
 
     paths.each do |p|
-      begin
-        basefilename = p.split('/').last
-        path = "#{cache_dir}/#{basefilename}"
-        data = download_file_cached "#{release_base_url}/#{p}", path
-        plainfile = "#{cache_dir}/Packages.plain"
-        File.open(plainfile, 'w') do |f|
-          case basefilename.downcase
-          when 'packages.xz'
-            require 'xz'
-            f << XZ.decompress(data)
-          when 'packages.gz'
-            require 'zlib'
-            f << Zlib.gunzip(data)
-          else
-            f << data
-          end
+      basefilename = p.split('/').last
+      path = "#{cache_dir}/#{basefilename}"
+      data = download_file_cached "#{release_base_url}/#{p}", path
+      plainfile = "#{cache_dir}/Packages.plain"
+      File.open(plainfile, 'w') do |f|
+        case basefilename.downcase
+        when 'packages.xz'
+          require 'xz'
```

Comment:
should we just require xz, zlib on top of the ruby file and just differentiate which compression algo to use?

Reply:
We can. The idea of doing it this way was to not require it, if no file with that compression was found.

Reply:
This would means, "ruby loads all requirements" which takes X sec, than, during runtime, it will recognize to load more libs and does run the code to load dependencies again which again takes Y sec.
I think, from the performance its better to load all at once.

Reply:
I have moved the requires to the top.

Except the ones that are only there for test-runs.
Which is all that comes after: `if $PROGRAM_NAME == __FILE__`